Sources/MessagingInApp/Views/InAppMessageView.swift

```diff
@@ -79,45 +91,52 @@ public class InAppMessageView: UIView {
             // EventBus callback function might not be on UI thread.
             // Switch to UI thread to update UI.
             Task { @MainActor in
-                self?.checkIfMessageAvailableToDisplay()
+                self?.refreshView()
             }
         }
     }
 
-    private func checkIfMessageAvailableToDisplay() {
+    // Updates the state of the View, if needed. Call as often as you need if an event happens that may cause the View to need to update.
+    private func refreshView(forceShowNextMessage: Bool = false) {
         guard let elementId = elementId else {
-            return
+            return // we cannot check if a message is available until element id set on View.
         }
 
         let queueOfMessagesForGivenElementId = localMessageQueue.getInlineMessages(forElementId: elementId)
-        let messageToDisplay = queueOfMessagesForGivenElementId.first
+        let messageAvailableToDisplay = queueOfMessagesForGivenElementId.first { potentialMessageToDisplay in
+            let didPreviouslyShowMessage = previouslyShownMessages.contains(where: { $0.id == potentialMessageToDisplay.id })
+
+            return !didPreviouslyShowMessage
+        }
```

Comment:
minor but can be simplified to read, i had to double check if what i am understanding is whats happenings

```
// Function to check if a message has been previously shown
func hasNotBeenPreviouslyShown(_ message: Message) -> Bool {
    return !previouslyShownMessages.contains { $0.id == message.id }
}

// Find the first message that has not been previously shown
let messageAvailableToDisplay = queueOfMessagesForGivenElementId.first(where: hasNotBeenPreviouslyShown)
```

Reply:
Done https://github.com/customerio/customerio-ios/pull/756/commits/c70e9691db346ba4dc16a5c0fff2bb209a724d18